Refactor: make whitelist validation a bit more robust

Wolfsblvt opened the pull request:
I was dumb and added a named host into my whitelist, which seems to not be able to be resolveable. I don't care much. But it broke all other whitelist entries.
Feel like having it a bit more robust and ignore invalid entries makes sense. Because why not. While debugging I already ended up with half of it.

## Copilot Summary
This pull request refactors the IP whitelist checking logic in the `getWhitelistMiddleware` function to improve error handling and code readability. The most important change is the introduction of a dedicated helper function for IP whitelist checks.

**Middleware logic improvements:**

* Added the `isIPInWhitelist` helper function to encapsulate IP whitelist checking, which improves error handling by catching and logging exceptions for individual whitelist entries and the overall check.
* Updated the main whitelist check to use the new helper function for both `clientIp` and `forwardedIp`, resulting in cleaner and more maintainable code.

<!-- Put X in the box below to confirm -->

## Checklist:

- [x] I have read the [Contribution guidelines](https://www.youtube.com/watch?v=e-GwojpaoQI).

Wolfsblvt committed:
refactor: extract IP whitelist validation into helper function

- Added isIPInWhitelist helper with error handling for individual whitelist entry checks
- Replaced inline whitelist matching logic with reusable function calls
- Added JSDoc type annotations and error logging for failed IP matching attempts

Wolfsblvt commented:
@codex review

Cohee1207 reviewed (commented):
There might be some console warning spam on every incoming request, which might softly nudge a user to check their whitelist. Is that okay/expected with you?

Wolfsblvt commented:
I saw that. I think that's okay and should be expected.
Better than crashing and not allowing any whitelist entries. It's a wrong configuration anyway. The alternative is to filter out invalid ones on app start, but eh.

Cohee1207 commented:
> The alternative is to filter out invalid ones on app start, but eh.

I thought of that. That seems more optimal to do once than every time on each request.

Wolfsblvt committed:
refactor: simplify whitelist validation with upfront filtering

- Moved IP validation to startup time instead of per-request checking
- Extracted validateWhitelist function to filter invalid entries once at initialization
- Simplified isIPInWhitelist by removing redundant error handling after validation

Wolfsblvt commented:
Done.

<img width="915" height="240" alt="image" src="https://github.com/user-attachments/assets/d8299249-44b3-4ae3-9468-b3d26892280a" />

@codex review again

Wolfsblvt committed:
fix: correct IP whitelist matching to use parsed CIDR notation

Wolfsblvt commented on src/middleware/whitelist.js:
```diff
@@ -91,9 +114,19 @@ export default async function getWhitelistMiddleware() {
 
+        /**
+         * Checks if an IP address matches any entry in the whitelist.
+         * @param {string[]} whitelist - The list of whitelisted IPs/CIDRs
+         * @param {string} ip - The IP address to check
+         * @returns {boolean} True if the IP matches any whitelist entry
+         */
+        function isIPInWhitelist(whitelist, ip) {
+            return whitelist.some(x => ipMatching.matches(ip, x));
+        }
+
         //clientIp = req.connection.remoteAddress.split(':').pop();
-        if (!whitelist.some(x => ipMatching.matches(clientIp, ipMatching.getMatch(x)))
-            || forwardedIp && !whitelist.some(x => ipMatching.matches(forwardedIp, ipMatching.getMatch(x)))
+        if (!isIPInWhitelist(whitelist, clientIp)
+            || forwardedIp && !isIPInWhitelist(whitelist, forwardedIp)
```
ahyo, missed that.
